template/context.go

```diff
@@ -107,9 +107,10 @@ type PayloadContext struct {
 	t                 *api.Task
 	restrictedSecrets exec.SecretGetter
 	restrictedConfigs exec.ConfigGetter
+	sensitive         bool
 }
 
-func (ctx PayloadContext) secretGetter(target string) (string, error) {
+func (ctx *PayloadContext) secretGetter(target string) (string, error) {
```

Comment:
I hadn't seen this recommendation before. I've changed all the methods to use pointer receivers.